pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -1059,6 +1091,202 @@ func testHTTPsGoTLSCaptureAlreadyRunningContainer(t *testing.T, cfg *config.Conf
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport %d", 44330)
+			},
+		},
+		{
+			name: "TLS_1_1_docker",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				go func() {
+					time.Sleep(5 * time.Second)
+					prototls.RunClientOpenssl(t, "localhost", "44330", "-tls1_1")
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport %d", 44330)
+			},
+		},
+		{
+			name: "TLS_1_2_docker",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				go func() {
+					time.Sleep(5 * time.Second)
+					prototls.RunClientOpenssl(t, "localhost", "44330", "-tls1_2")
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport %d", 44330)
+			},
+		},
+		{
+			name: "TLS_1_3_docker",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				go func() {
+					time.Sleep(5 * time.Second)
+					prototls.RunClientOpenssl(t, "localhost", "44330", "-tls1_3")
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport %d", 44330)
+			},
+		},
```

Comment:
there's a huge duplication here, let's simplify it
an example in kafka classification tests (https://github.com/DataDog/datadog-agent/blob/main/pkg/network/tracer/tracer_classification_test.go#L372-L415) where we are running the same test for different kafka versions

Reply:
I not sure there are a value to factorize tests as in some case it make it more complicated to what was failing and why not the previous one. This could be counter intuitive too
I did factorize for the different tls version but still need to add exception for the last test

Reply:
looks much better now